tests/test_edgeql_ddl.py

```diff
@@ -10508,6 +10508,115 @@ async def test_edgeql_ddl_link_policy_09(self):
             DELETE Tgt;
         """)
 
+    async def test_edgeql_ddl_link_policy_10(self):
+        # Make sure we NULL out the pointer on the delete, which will
+        # trigger the constraint
+        await self.con.execute(r"""
+            SET MODULE test;
+
+            CREATE TYPE Tgt;
+            CREATE TYPE Foo {
+                CREATE LINK tgt -> Tgt {
+                    ON TARGET DELETE ALLOW;
+                };
+                CREATE CONSTRAINT expression on (EXISTS .tgt);
+
+            };
+            CREATE TYPE Bar EXTENDING Foo;
+        """)
+
+        await self.con.execute(r"""
+            INSERT Bar { tgt := (INSERT Tgt) };
+        """)
+
+        async with self.assertRaisesRegexTx(
+            edgedb.ConstraintViolationError,
+            'invalid Bar',
+        ):
+            await self.con.execute("""
+                DELETE Tgt;
+            """)
+
+        await self.con.execute("""
+            DELETE Bar;
+            DELETE Tgt;
+        """)
+
+    @test.xfail('''
+        We can't properly compile the constraint here in the basic case,
+        let alone combined with DELETE ALLOW.
+
+        We probably should be rejecting the constraint.
+    ''')
+    async def test_edgeql_ddl_link_policy_11(self):
+        # Should we even be allowing (EXISTS .tgt) as a constraint?
```

Comment:
I see no reason why it couldn't be equivalent to `required`, but I'm also OK with punting this and prohibiting `EXISTS` along with other references to `multi` on object constraints.